app/controllers/posts_controller.rb

```diff
@@ -26,12 +26,24 @@ def show
     @comments = Comment.where(post_id: params[:id])
   end
 
+  def destroy
+    @user = User.find(params[:user_id])
+    @post = @user.posts.find(params[:id])
+    authorize! :destroy, @post
+    @post.destroy
+    redirect_to user_posts_path(current_user, @post), notice: 'Post was successfully destroyed.'
+  end
```

Comment:
- [ ] Great job adding the ability class to implement the authorization of users, but there is work to be done. I created two accounts using different browsers, I was able to delete a post and a comment created by the first user which should not be the case. I also noticed that one user (first) was creating posts this behavior should be fixed as well. Kindly make changes and submit it for another review. You can use `load_and_authorize_resource` in the Post controller for __CanCanCan__ to work properly.

Reply:
Hello @elisha2kyakpo1 , thanks for the observation. The project has been updated to fix the issue.